unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2597,6 +2598,79 @@ bool includeInvisible
             actionFinished(true);
         }
 
+        public void Get2DSemanticHulls(List<string> objectTypes = null) {
+            HashSet<string> allowedObjectTypesSet = null;
+            if (objectTypes != null) {
+                allowedObjectTypesSet = new HashSet<string>(objectTypes);
+            }
+
+            var objectIdToConvexHull = new Dictionary<string, List<(float x, float y)>>();
+            foreach (var sop in GameObject.FindObjectsOfType<SimObjPhysics>()) {
+                if (
+                    allowedObjectTypesSet != null 
+                    && !allowedObjectTypesSet.Contains(sop.Type.ToString())
+                ) {
+                    continue;
+                }
+                var vertices = new List<MIConvexHull.DefaultVertex2D>();
+                float maxY = -float.PositiveInfinity;
+
+                foreach (var meshFilter in sop.GetComponentsInChildren<MeshFilter>()) {
+                    foreach (var localVertex in meshFilter.mesh.vertices) {
+                        var globalVertex = meshFilter.transform.TransformPoint(localVertex);
+                        vertices.Add(new MIConvexHull.DefaultVertex2D(globalVertex.x, globalVertex.z));
+                        maxY = Math.Max(maxY, globalVertex.y);
+                    }
+                }
+                var miconvexHull = MIConvexHull.ConvexHull.Create2D(vertices, 1e-10);
```

Comment:
```c#
var miconvexHull = MIConvexHull.ConvexHull.Create2D(vertices, 1e-10);
```
becomes something like
```c#
ConvexHullCreationResult miconvexHull = MIConvexHull.ConvexHull.Create2D(
    data: vertices,
    tolerance: 1e-10
);
```
(perhaps it's `ConvexHullCreationResult<TVertex>` instead of `ConvexHullCreationResult`:
https://github.com/DesignEngrLab/MIConvexHull/blob/241b1179e63d12e80ea689436bc37045553a53eb/MIConvexHull/ConvexHull.cs#L98)

Reply:
Done.